bridger-cli/src/utils/transactionHandler.ts

```diff
@@ -0,0 +1,159 @@
+    const estimateGas = await this.veaOutbox.estimateGas.claim(this.epoch, stateRoot, { value: bridgeConfig.deposit });
+    const claimTransaction = await this.veaOutbox.claim(this.epoch, stateRoot, {
+      value: bridgeConfig.deposit,
+      gasLimit: estimateGas,
+    });
+    console.log(`Epoch ${this.epoch} was claimed with trnx hash ${claimTransaction.hash}`);
+    this.pendingTransactions.claim = claimTransaction.hash;
+  }
+
+  public async startVerification(latestBlockTimestamp: number) {
+    if (this.claim == null) {
+      throw new Error("Claim is not set");
+    }
+    if (await this.checkTransactionPendingStatus(this.pendingTransactions.startVerification)) {
+      console.log(
+        "Start verification transaction is still pending with hash: " + this.pendingTransactions.startVerification
+      );
+      return;
+    }
+    const bridgeConfig = this.getBridgeConfig(this.chainId);
+    const timeOver =
+      latestBlockTimestamp - this.claim.timestampClaimed - bridgeConfig.sequencerDelayLimit - bridgeConfig.epochPeriod;
+    console.log(timeOver);
+    if (timeOver >= 0) {
+      const estimateGas = await this.veaOutbox.estimateGas.startVerification(this.epoch, this.claim);
+      const startVerifTrx = await this.veaOutbox.startVerification(this.epoch, this.claim, { gasLimit: estimateGas });
+      console.log(`Verification started for epoch ${this.epoch} with trx hash ${startVerifTrx.hash}`);
+      this.pendingTransactions.startVerification = startVerifTrx.hash;
+    } else {
+      console.log("Sequencer delay not passed yet, seconds left: " + -1 * timeOver);
+    }
+  }
+
+  public async verifySnapshot(latestBlockTimestamp: number) {
+    if (this.claim == null) {
+      throw new Error("Claim is not set");
+    }
+    if (await this.checkTransactionPendingStatus(this.pendingTransactions.verifySnapshot)) {
+      console.log("Verify snapshot transaction is still pending with hash: " + this.pendingTransactions.verifySnapshot);
+      return;
+    }
+    const bridgeConfig = this.getBridgeConfig(this.chainId);
+
+    const timeLeft = latestBlockTimestamp - this.claim.timestampClaimed - bridgeConfig.minChallengePeriod;
+    console.log("Time left for verification: " + timeLeft);
+    console.log(latestBlockTimestamp, this.claim.timestampClaimed, bridgeConfig.minChallengePeriod);
+    // Claim not resolved yet, check if we can verifySnapshot
+    if (timeLeft >= 0) {
+      console.log("Verification period passed, verifying snapshot");
+      // Estimate gas for verifySnapshot
+      const estimateGas = await this.veaOutbox.estimateGas.verifySnapshot(this.epoch, this.claim);
+      const claimTransaction = await this.veaOutbox.verifySnapshot(this.epoch, this.claim, {
+        gasLimit: estimateGas,
+      });
+      console.log(`Epoch ${this.epoch} verification started with trnx hash ${claimTransaction.hash}`);
+      this.pendingTransactions.verifySnapshot = claimTransaction.hash;
+    } else {
```

Comment:
(ignoreable) You can probably convert this if/else into a guard and simplify the branching.

```ts

if (timeleft < 0) {
  console.log("Censorship test in progress, sec left: " + -1 * timeLeft);
  return;
}

// Here goes the current content of if (timeLeft >= 0) branch
```